frontend/src/metabase/common/components/AggregationPicker/AggregationPicker.tsx

```diff
@@ -68,28 +63,34 @@ function isOperatorListItem(item: ListItem): item is OperatorListItem {
 export function AggregationPicker({
   className,
   query,
+  clause,
   stageIndex,
   operators,
   hasExpressionInput = true,
-  legacyQuery,
-  legacyClause,
   maxHeight = DEFAULT_MAX_HEIGHT,
   onSelect,
-  onSelectLegacy,
   onClose,
 }: AggregationPickerProps) {
+  const metadata = useSelector(getMetadata);
+  const displayInfo = clause
+    ? Lib.displayInfo(query, stageIndex, clause)
+    : undefined;
+  const initialOperator = getInitialOperator(query, stageIndex, operators);
   const [
     isEditingExpression,
     { turnOn: openExpressionEditor, turnOff: closeExpressionEditor },
-  ] = useToggle(isExpressionEditorInitiallyOpen(legacyClause));
+  ] = useToggle(isExpressionEditorInitiallyOpen(displayInfo, initialOperator));
+  const datasetQuery = Lib.toLegacyQuery(query);
+  const legacyQuery = new StructuredQuery(
```

Comment:
IDK if it's a case here, but you can get pre MLv2 query wiht `query.question()._getMLv2Query()`